packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,62 +7,106 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
+import {TDateRange} from './types';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	dateRange: TDateRange;
 	disabled?: boolean;
-	onClick: (date: Date) => void;
+	enableDateRange?: boolean;
+	onClick: (date: Date, enableDateRange?: boolean) => void;
+}
+
+interface IInterval {
+	start: Date;
+	end: Date;
+}
+
+function isWithinInterval(dirtyDate: Date, dirtyInterval: IInterval) {
+	const interval = dirtyInterval || {};
+	const time = dirtyDate.getTime();
+	const startTime = interval.start.getTime();
+	const endTime = interval.end.getTime();
+
+	if (!(startTime <= endTime)) {
+		return false;
+	}
+
+	return time >= startTime && time <= endTime;
 }
 
 const ClayDatePickerDayNumber: React.FunctionComponent<IProps> = ({
+	dateRange,
 	day,
-	daySelected,
 	disabled,
+	enableDateRange,
 	onClick,
 }) => {
 	const {date, outside} = day;
 
+	const isSelectedToDate =
+		date.toDateString() === dateRange.toDate.toDateString();
+	const isSelectedFromDate =
+		date.toDateString() === dateRange.fromDate.toDateString();
+
 	const classNames = classnames(
 		'date-picker-date date-picker-calendar-item',
 		{
-			active: date.toDateString() === daySelected.toDateString(),
+			active: isSelectedFromDate || (enableDateRange && isSelectedToDate),
 			disabled: outside || disabled,
 		}
 	);
 
 	return (
-		<button
-			aria-label={formatDate(
-				setDate(date, {
+		<div
+			className={classnames(
+				'date-picker-col',
+				enableDateRange && {
+					'c-selected':
+						!isSelectedFromDate &&
+						!isSelectedToDate &&
+						isWithinInterval(date, {
+							end: dateRange.toDate,
+							start: dateRange.fromDate,
+						}),
+					'c-selected c-selected-end':
+						isSelectedToDate &&
+						!(isSelectedToDate && isSelectedFromDate),
+					'c-selected c-selected-start':
+						isSelectedFromDate &&
+						!(isSelectedToDate && isSelectedFromDate),
+				}
+			)}
+		>
+			<button
+				aria-label={setDate(date, {
```

Comment:
Before this change the day selection button was printing a string like `2012 12 21` instead of `2012/12/21`.
Instead of `toLocaleDateString()` I could use `toDateString()` and the label will be like: `Thusday Dec 12 2021`. What do you think @jbalsas @marcoscv-work @edalgrin ?

Reply:
Sounds reasonable, @diegonvs!